src/kv/Makefile.am

```diff
@@ -16,12 +16,17 @@ if WITH_SLIBROCKSDB
 # build rocksdb with its own makefile
 # for some stupid reason this needs -fPIC...
 # PORTABLE=1 fixes the aarch64 build (-march=native doesn't work there)
-NPROC = nproc
+
+ROCKSDBCXX_FLAGS = "-fPIC"
 if FREEBSD
-        NPROC = sysctl -n hw.ncpu
+NPROC = sysctl -n hw.ncpu
+ROCKSDBCXX_FLAGS += "-Wmismatched-tags"
+else
+NPROC = nproc
 endif
+
 rocksdb/librocksdb.a:
-	cd rocksdb && CC="${CC}" CXX="${CXX}" EXTRA_CXXFLAGS=-fPIC PORTABLE=1 $(MAKE) -j$(shell ${NPROC}) static_lib
+	cd rocksdb && CC="${CC}" CXX="${CXX}" EXTRA_CXXFLAGS="${ROCKSDBCXX_FLAGS}" PORTABLE=1 ${MAKE} -j$(shell ${NPROC}) static_lib
```

Comment:
@trociny 
You are right...
Reason I changed it a bit is because in the current implementation nproc is always executed, even if not available. And I think that gmake even stops then.
So even the linux part needs to be conditionalised, not to break on FreeBSD.
But I'll fix the commit message to call it a rearange?